src/main/java/uk/gov/hmcts/reform/sandl/snlapi/exceptions/OptimisticLockException.java

```diff
@@ -1,12 +1,10 @@
 package uk.gov.hmcts.reform.sandl.snlapi.exceptions;
 
+import lombok.NoArgsConstructor;
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
 @ResponseStatus(value = HttpStatus.CONFLICT, reason = "The object version has changed")
+@NoArgsConstructor
 public class OptimisticLockException extends RuntimeException {
```

Comment:
is this implementation following the best practices for writing custom exceptions in java?

Reply:
https://stackify.com/java-custom-exceptions/
I will add documentation, which will also explain why there is only one constructor - shortly spring usage of @ResponseStatus and yes this is tightly coupled to spring.